packages/integration-tests/src/EntityManager.test.ts

```diff
@@ -1023,6 +1023,95 @@ describe("EntityManager", () => {
     // Then the deleted entity was returned from the flush
     expect(result).toEqual(a1);
   });
+
+  it("can clone entities", async() => {
+    const em = newEntityManager();
+
+    // Given an entity
+    const a1 = new Author(em, {firstName: "a1"});
+    await em.flush();
+
+    // When we clone that entity
+    const a2 =  await em.cloneEntity(a1);
+
+    // Then we expect the cloned entity to have the same properties as the original
+    expect(a2.firstName).toEqual(a1.firstName);
+    expect(a2.id).not.toEqual(a1.id)
+  });
+
+  it("can clone entities and referenced entities", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity
+    const a1 = new Author(em, {firstName: "a1"});
+    const b1 = new Book(em, {title: "b1", author: a1});
+    await em.flush();
+
+    // When we clone that entity and its reference
+    const a2 =  await em.cloneEntity(a1, "books");
+    await em.flush();
+
+    // Then we expect the cloned entity to have a cloned copy of the original's reference
+    expect(a2.books.get[0].title).toEqual(b1.title);
+    expect(a2.books.get[0].id).not.toEqual(b1.id);
+  });
+
+  it("cannot clone many-to-many references", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity with a many-to-many reference
+    const a1 = new Author(em, {firstName: "a1"});
+    const b1 = new Book(em, {title: "b1", author: a1});
+    const t1 = new Tag(em, {name: "t1", books: [b1]})
+    await em.flush();
+
+    // When we clone that entity and its nested references
+    const toExecute = async function () { await em.cloneEntity(a1, {books: "tags"})};
+
+    // Then we expect the cloning to fail
+    await expect(toExecute).rejects.toThrow()
```

Comment:
nit: Can you include the message which is thrown please?
```suggestion
    await expect(toExecute).rejects.toThrow("some sort of error")
```

Reply:
+1